specs/src/specs/data_structures.md

```diff
@@ -497,31 +497,31 @@ For shares **with a reserved namespace ID through [`NAMESPACE_ID_MAX_RESERVED`](
 
 > **Note** The first [`NAMESPACE_ID_BYTES`](./consensus.md#constants) of a share's raw data `rawData` is the namespace ID of that share, `namespaceID`. The next [`SHARE_INFO_BYTES`](./consensus.md#constants) bytes are for share information.
 
-- If this is the first share of a sequence, the next 1 to 10 bytes contain a [varint](https://developers.google.com/protocol-buffers/docs/encoding) of the `uint64` length of the sequence that follows, in bytes.
+- If this is the first share of a sequence, the next [`SEQUENCE_BYTES`](./consensus.md#constants) contain a big endian `uint32` that represents the length of the sequence that follows in bytes.
```

Comment:
```suggestion
- If this is the first share of a sequence, the next [`SEQUENCE_BYTES`](./consensus.md#constants) contain a big endian `uint64` that represents the length of the sequence that follows in bytes.
```

Building on Evan's comment above, I think this needs to be updated to match right? Seems like the table references `uint64` but all the explanations here reference `uint32`

Reply:
Sorry this is a bit confusing. The two types are for two different fields. 
1. The constant `SEQUENCE_BYTES` is an a hard-coded number: `4`. The type of this constant can be `uint8`, `uint16`, `uint32`, or `uint64` because all four types can contain the number `4`. In practice, I choose `uint64` to match some of the other constants in that table.
2. The value written to the `4` sequence bytes is a `uint32` and it is variable (i.e. the sequence length is based on the sequence it is describing). Since a `uint64` would occupy `8` bytes, it can't be written to the sequence bytes. We aligned on a `uint32` for the sequence length in https://github.com/celestiaorg/celestia-app/blob/main/docs/architecture/adr-012-sequence-length-encoding.md

Maybe this comment helps: https://github.com/celestiaorg/celestia-app/pull/1124#discussion_r1049838170